server/cloudformation/environment_test.go

```diff
@@ -0,0 +1,475 @@
+package cloudformation
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/remind101/empire"
+	"github.com/remind101/empire/empiretest"
+	"github.com/remind101/empire/pkg/image"
+	"github.com/remind101/empire/scheduler"
+	"github.com/remind101/pkg/timex"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+var fakeNow = time.Date(2015, time.January, 1, 1, 1, 1, 1, time.UTC)
+
+// Stubs out time.Now in empire.
+func init() {
+	timex.Now = func() time.Time {
+		return fakeNow
+	}
+}
+
+// Run the tests with empiretest.Run, which will lock access to the database
+// since it can't be shared by parallel tests.
+func TestMain(m *testing.M) {
+	empiretest.Run(m)
+}
+
+func TestProvision_Create(t *testing.T) {
```

Comment:
I think inside this package, we should just unit test the struct (mock out empire.Empire). Some actual integration level tests for the cloudformation resources would be nice, but we should put integration level tests in `./tests`.